instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py

```diff
@@ -256,6 +257,19 @@ def _trace_failure(*args, **kwargs):
             return
 
         if ex is not None:
+            # Unwrap the actual exception wrapped by billiard's
+            # `ExceptionInfo` and `ExceptionWithTraceback`.
+            if (
+                isinstance(ex, ExceptionInfo)
+                and ex.exception is not None
+            ):
+                ex = ex.exception
+            elif (
+                isinstance(ex, ExceptionWithTraceback)
```

Comment:
I think here you meant to check isinstance(ex.exception, ExceptionWithTraceback)
and ex.exception.exc is not None
because the ex is the ExceptionInfo the ex.exception is the ExceptionWithTraceback, right?

Reply:
Is that how it goes? I'm most likely misunderstanding how Celery wraps these exceptions. (It certainly doesn't help that I don't understand _why_ Celery wraps these exceptions)

I thought what you were saying earlier was that `ex` was either an `ExceptionInfo` in older versions (and therefore we unwrap `ex.exception`) or an `ExceptionWithTraceback` in newer versions (and therefore we unwrap `ex.exc`) -- but the way you describe it now, it seems that an `ExceptionInfo` wraps an `ExceptionWithTraceback` which wraps the actual exception? 🤔 

Let me know if I understood it right this time and I'll amend it. To be clear, in that case, would the twice-wrapped exception (`ex.exception.exc`) have a type and traceback that are meaningful to the user? Or should we somehow grab the traceback from `ExceptionWithTraceback` instead?

Reply:
Done! I also made it so that the import of `ExceptionWithTraceback` is allowed to fail, for older versions of Celery or billiards where it doesn't exist.